sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java

```diff
@@ -362,82 +362,109 @@
     .setDomain(DOMAIN_COMPLEXITY)
     .create();
 
+  /**
+   * @deprecated since 6.7
+   */
+  @Deprecated
   public static final String FILE_COMPLEXITY_KEY = "file_complexity";
-
   /**
    * Information about the cyclomatic complexity per file, calculated by divided the {@link #COMPLEXITY} by the number of {@link #FILES}.
+   *
+   * @deprecated since 6.7
    */
+  @Deprecated
   public static final Metric<Double> FILE_COMPLEXITY = new Metric.Builder(FILE_COMPLEXITY_KEY, "Complexity / File", Metric.ValueType.FLOAT)
     .setDescription("Complexity average by file")
     .setDirection(Metric.DIRECTION_WORST)
     .setQualitative(true)
     .setDomain(DOMAIN_COMPLEXITY)
+    .setHidden(true)
     .create();
 
   /**
    * @since 3.6
+   * @deprecated since 6.7
    */
+  @Deprecated
   public static final String COMPLEXITY_IN_CLASSES_KEY = "complexity_in_classes";
 
   /**
    * @since 3.6
+   * @deprecated since 6.7
    */
+  @Deprecated
   public static final Metric<Integer> COMPLEXITY_IN_CLASSES = new Metric.Builder(COMPLEXITY_IN_CLASSES_KEY, "Complexity in Classes", Metric.ValueType.INT)
     .setDescription("Cyclomatic complexity in classes")
     .setHidden(true)
     .setDirection(Metric.DIRECTION_WORST)
     .setQualitative(false)
     .setDomain(DOMAIN_COMPLEXITY)
     .setDeleteHistoricalData(true)
+    .setHidden(true)
     .create();
 
+  /**
+   * @deprecated since 6.7
+   */
+  @Deprecated
   public static final String CLASS_COMPLEXITY_KEY = "class_complexity";
-
   /**
    * Information about the cyclomatic complexity per class, calculated by divided the {@link #COMPLEXITY_IN_CLASSES} by the number of {@link #CLASSES}.
+   * @deprecated since 6.7
    */
+  @Deprecated
   public static final Metric<Double> CLASS_COMPLEXITY = new Metric.Builder(CLASS_COMPLEXITY_KEY, "Complexity / Class", Metric.ValueType.FLOAT)
     .setDescription("Complexity average by class")
     .setDirection(Metric.DIRECTION_WORST)
     .setQualitative(true)
     .setDomain(DOMAIN_COMPLEXITY)
+    .setHidden(true)
     .create();
 
   /**
    * @since 3.6
+   * @deprecated since 6.7
    */
+  @Deprecated
   public static final String COMPLEXITY_IN_FUNCTIONS_KEY = "complexity_in_functions";
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)